tests/z_api_null_drop_test.c

```diff
@@ -155,6 +161,7 @@ int main(int argc, char **argv) {
         z_drop(&closure_zid_null_2);
         z_drop(&reply_channel_closure_null_2);
         z_drop(&reply_channel_null_2);
+        z_drop(&str_null_2);
```

Comment:
In Zenoh-Pico, we have it implemented as `z_drop(z_move(str_null_2))`.
I know it might seem redundant but by doing so we dont break the rule of being explicit when moving things to the Zenoh API.